app/src/org/commcare/tasks/CleanRawMedia.kt

```diff
@@ -0,0 +1,30 @@
+package org.commcare.tasks
+
+import android.content.Context
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import org.commcare.CommCareApplication
+import org.commcare.android.database.user.models.FormRecord
+import org.commcare.utils.CrashUtil
+import java.io.File
+import java.lang.Exception
+
+class CleanRawMedia(appContext: Context, workerParams: WorkerParameters)
+    : Worker(appContext, workerParams) {
+
+    override fun doWork(): Result {
+        try {
+            val storage = CommCareApplication.instance().getUserStorage(FormRecord::class.java)
+            for (formRecord in storage) {
```

Comment:
I think we should only be doing this for Forms in a specific state, right? It doesn't seem correct to me for this to delete the raw storage for, say, a form that's currently in progress. I wonder if it's safest to, say, only do this for forms that were saved more than a few days ago, too, but that's probably overly conservative

Reply:
Yeah thats a good point. Though given on some devices users face storage issues even in normal conditions, I am inclined to remove it for all completely saved forms. I don't see value in retaining the original images for any forms which are not in progress or incomplete.